nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/api/Framework.java

```diff
@@ -472,10 +479,14 @@ public static boolean isDevModeSet() {
      * Activating this mode, some of the code may not behave as it would in production, to ease up testing.
      */
     public static boolean isTestModeSet() {
-        if (testModeSet == null) {
-            testModeSet = isBooleanPropertyTrue(NUXEO_TESTING_SYSTEM_PROP);
+        String testModeCacheDisabled = System.getProperty(NUXEO_TESTING_PROP_CACHE_DISABLED);
+        if (Boolean.FALSE.equals(Boolean.parseBoolean(testModeCacheDisabled))) {
```

Comment:
Note that `parseBoolean` returns a `boolean`, not a `Boolean`.
But you can use `Boolean.getBoolean(syspropname)` instead of `System.getProperty` + `Boolean.parseBoolean`